blockchain/blockchain.go

```diff
@@ -67,23 +71,82 @@ type BlockResult struct {
 	TotalGas uint64
 }
 
-// UpdateGasPriceAvg Updates the rolling average value of the gas price
-func (b *Blockchain) UpdateGasPriceAvg(newValue *big.Int) {
-	b.agpMux.Lock()
+// updateGasPriceAvg updates the rolling average value of the gas price
+func (b *Blockchain) updateGasPriceAvg(newValues []*big.Int) {
+	b.gpAverage.Lock()
+	defer b.gpAverage.Unlock()
+
+	//	Sum the values for quick reference
+	sum := big.NewInt(0)
+	for _, val := range newValues {
+		sum = sum.Add(sum, val)
+	}
+
+	// There is no previous average data,
+	// so this new value set will instantiate it
+	if b.gpAverage.count.Uint64() == 0 {
+		b.getArithmeticAverage(newValues, sum)
+
+		return
+	}
+
+	// There is existing average data,
+	// use it to generate a new average
+	b.getRollingAverage(newValues, sum)
+}
+
+// getArithmeticAverage calculates and sets the arithmetic average
+// of the passed in data set
+func (b *Blockchain) getArithmeticAverage(newValues []*big.Int, sum *big.Int) {
```

Comment:
why get...? This method updates its own state.

Reply:
You're completely right, it's leftover 🤦 

Fixed: 

[34498f1](https://github.com/0xPolygon/polygon-edge/pull/483/commits/34498f1e86e3847fbf343bc4b87d7ca430e20ca8)